src/form/FileUpload/hooks.ts

```diff
@@ -64,19 +69,21 @@ export const useFileUpload = (
 
 				xhr.onreadystatechange = () => {
 					if (xhr.readyState !== XMLHttpRequest.DONE) return;
+
 					const status = xhr.status;
+					const response = xhr.responseText;
 
-					if (status === 0 || (status >= 200 && status < 400)) {
+					if (status >= 200 && status < 400) {
 						// The request has been completed successfully
 						customFile.progress = 100;
-						customFile.response = xhr.responseText;
+						customFile.response = response;
 						customFile.uploadXhrError = undefined;
 						// Let application track uploaded files
 						onFileSuccess && onFileSuccess(customFile);
 					} else {
 						// Something went wrong with the request
```

Comment:
Does this mean that onreadystatechange is never called with the default `0` status without it being an error?

Reply:
No the state change is called on every state change, we only handle the state change when it has the state DONE.

https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/status